# A/B Testing to Find the Best Marketing Strategy

# Control Group vs Test Group

### In marketing campaigns, marketers send campaign ads to selected customers ("test group") and putting aside the randomly selected some group of customers ("control group") from the campaign. The idea is to determine the effectiveness of the compaign by comparing the sales generated from the "test group" and "control group".

### For example, an ecommerce website has segmented customers to send "50% discount on clothing" campaign and they send this campaign to only 50k of them (“test group”), put aside a randomly-selected 5k customers (“control group”) who will not receive it. Once the campaign is over, they will determine the effectiveness of the campaign by comparing the additional revenues generated by the test group with those generated by the control group.

### It can also be used for comparing 2 different market Strategy.

In [77]:
import plotly.express as px
import pandas as pd

In [2]:
df_Test = pd.read_csv("Test Group.csv")
df_Test

Campaign Name        Date  Spend [USD]  Number of Impressions   Reach  \
0   Test Campaign  01-08-2019         3008                  39550   35820   
1   Test Campaign  02-08-2019         2542                 100719   91236   
2   Test Campaign  03-08-2019         2365                  70263   45198   
3   Test Campaign  04-08-2019         2710                  78451   25937   
4   Test Campaign  05-08-2019         2297                 114295   95138   
5   Test Campaign  06-08-2019         2458                  42684   31489   
6   Test Campaign  07-08-2019         2838                  53986   42148   
7   Test Campaign  08-08-2019         2916                  33669   20149   
8   Test Campaign  09-08-2019         2652                  45511   31598   
9   Test Campaign  10-08-2019         2790                  95054   79632   
10  Test Campaign  11-08-2019         2420                  83633   71286   
11  Test Campaign  12-08-2019         2831                 124591   10598   
12  Test Campaign  13-08-2019         1972                  65827   49531   
13  Test Campaign  14-08-2019         2537                  56304   25982   
14  Test Campaign  15-08-2019         2516                  94338   76219   
15  Test Campaign  16-08-2019         3076                 106584   81389   
16  Test Campaign  17-08-2019         1968                  95843   54389   
17  Test Campaign  18-08-2019         1979                  53632   43241   
18  Test Campaign  19-08-2019         2626                  22521   10698   
19  Test Campaign  20-08-2019         2712                  39470   31893   
20  Test Campaign  21-08-2019         3112                 133771  109834   
21  Test Campaign  22-08-2019         2899                  34752   27932   
22  Test Campaign  23-08-2019         2407                  60286   49329   
23  Test Campaign  24-08-2019         2078                  36650   30489   
24  Test Campaign  25-08-2019         2928                 120576  105978   
25  Test Campaign  26-08-2019         2311                  80841   61589   
26  Test Campaign  27-08-2019         2915                 111469   92159   
27  Test Campaign  28-08-2019         2247                  54627   41267   
28  Test Campaign  29-08-2019         2805                  67444   43219   
29  Test Campaign  30-08-2019         1977                 120203   89380   

    Number of Website Clicks  Number of Searches  Number of View Content  \
0                       3038                1946                    1069   
1                       4657                2359                    1548   
2                       7885                2572                    2367   
3                       4216                2216                    1437   
4                       5863                2106                     858   
5                       7488                1854                    1073   
6                       4221                2733                    2182   
7                       7184                2867                    2194   
8                       8259                2899                    2761   
9                       8125                2312                    1804   
10                      3750                2893                    2617   
11                      8264                2081                    1992   
12                      7568                2213                    2058   
13                      3993                1979                    1059   
14                      4993                2537                    1609   
15                      6800                2661                    2594   
16                      7910                1995                    1576   
17                      6909                2824                    2522   
18                      7617                2924                    2801   
19                      6050                2061                    1894   
20                  

In [3]:
df_Control = pd.read_csv("Control Group.csv")
df_Control

Campaign Name        Date  Spend [USD]  Number of Impressions  \
0   Control Campaign  01-08-2019         2280                82702.0   
1   Control Campaign  02-08-2019         1757               121040.0   
2   Control Campaign  03-08-2019         2343               131711.0   
3   Control Campaign  04-08-2019         1940                72878.0   
4   Control Campaign  05-08-2019         1835                    NaN   
5   Control Campaign  06-08-2019         3083               109076.0   
6   Control Campaign  07-08-2019         2544               142123.0   
7   Control Campaign  08-08-2019         1900                90939.0   
8   Control Campaign  09-08-2019         2813               121332.0   
9   Control Campaign  10-08-2019         2149               117624.0   
10  Control Campaign  11-08-2019         2490               115247.0   
11  Control Campaign  12-08-2019         2319               116639.0   
12  Control Campaign  13-08-2019         2697                82847.0   
13  Control Campaign  14-08-2019         1875               145248.0   
14  Control Campaign  15-08-2019         2774               132845.0   
15  Control Campaign  16-08-2019         2024                71274.0   
16  Control Campaign  17-08-2019         2177               119612.0   
17  Control Campaign  18-08-2019         1876               108452.0   
18  Control Campaign  19-08-2019         2596               107890.0   
19  Control Campaign  20-08-2019         2675               113430.0   
20  Control Campaign  21-08-2019         1803                74654.0   
21  Control Campaign  22-08-2019         2939               105705.0   
22  Control Campaign  23-08-2019         2496               129880.0   
23  Control Campaign  24-08-2019         1892                72515.0   
24  Control Campaign  25-08-2019         1962               117006.0   
25  Control Campaign  26-08-2019         2233               124897.0   
26  Control Campaign  27-08-2019         2061               104678.0   
27  Control Campaign  28-08-2019         2421               141654.0   
28  Control Campaign  29-08-2019         2375                92029.0   
29  Control Campaign  30-08-2019         2324               111306.0   

       Reach  Number of Website Clicks  Number of Searches  \
0    56930.0                    7016.0              2290.0   
1   102513.0                    8110.0              2033.0   
2   110862.0                    6508.0              1737.0   
3    61235.0                    3065.0              1042.0   
4        NaN                       NaN                 NaN   
5    87998.0                    4028.0              1709.0   
6   127852.0                    2640.0              1388.0   
7    65217.0                    7260.0              3047.0   
8    94896.0                    6198.0              2487.0   
9    91257.0                    2277.0              2475.0   
10   95843.0                    8137.0              2941.0   
11  100189.0                    2993.0              1397.0   
12   68214.0                    6554.0              2390.0   
13  118632.0                    4521.0              1209.0   
14  102479.0                    4896.0              1179.0   
15   42859.0                    5224.0              2427.0   
16  106518.0                    6628.0              1756.0   
17   96518.0                    7253.0              2447.0   
18   81268.0                    3706.0              2483.0   
19   78625.0                    2578.0              1001.0   
20   59873.0                    5691.0              2711.0   
21   86218.0                    6843.0              3102.0   
22  109413.0                    4410.0              2896.0   
23   51987.0                    4085.0              1274.0   
24  100398.0                    4234.0              2423.0   
25   98432.0                    5435.0              2847.0   
26   91579.0                    4941.0              3549.0   
27  125874.0                    6287.0

In [4]:
#  Checking for NULL values
df_Test.isna().sum()

Campaign Name               0
Date                        0
Spend [USD]                 0
Number of Impressions       0
Reach                       0
Number of Website Clicks    0
Number of Searches          0
Number of View Content      0
Number of Add to Cart       0
Number of Purchase          0
dtype: int64

In [5]:
#  Checking for NULL values
df_Control.isna().sum()

Campaign Name               0
Date                        0
Spend [USD]                 0
Number of Impressions       1
Reach                       1
Number of Website Clicks    1
Number of Searches          1
Number of View Content      1
Number of Add to Cart       1
Number of Purchase          1
dtype: int64

In [6]:
# Changing data type
for column in df_Control.columns:
    if df_Control[column].dtype == 'float64':
        df_Control[column] = df_Control[column].fillna(df_Control[column].mean())
df_Control.dtypes

print(df_Control.isna().sum())
df_Control

Campaign Name               0
Date                        0
Spend [USD]                 0
Number of Impressions       0
Reach                       0
Number of Website Clicks    0
Number of Searches          0
Number of View Content      0
Number of Add to Cart       0
Number of Purchase          0
dtype: int64


Campaign Name        Date  Spend [USD]  Number of Impressions  \
0   Control Campaign  01-08-2019         2280           82702.000000   
1   Control Campaign  02-08-2019         1757          121040.000000   
2   Control Campaign  03-08-2019         2343          131711.000000   
3   Control Campaign  04-08-2019         1940           72878.000000   
4   Control Campaign  05-08-2019         1835          109559.758621   
5   Control Campaign  06-08-2019         3083          109076.000000   
6   Control Campaign  07-08-2019         2544          142123.000000   
7   Control Campaign  08-08-2019         1900           90939.000000   
8   Control Campaign  09-08-2019         2813          121332.000000   
9   Control Campaign  10-08-2019         2149          117624.000000   
10  Control Campaign  11-08-2019         2490          115247.000000   
11  Control Campaign  12-08-2019         2319          116639.000000   
12  Control Campaign  13-08-2019         2697           82847.000000   
13  Control Campaign  14-08-2019         1875          145248.000000   
14  Control Campaign  15-08-2019         2774          132845.000000   
15  Control Campaign  16-08-2019         2024           71274.000000   
16  Control Campaign  17-08-2019         2177          119612.000000   
17  Control Campaign  18-08-2019         1876          108452.000000   
18  Control Campaign  19-08-2019         2596          107890.000000   
19  Control Campaign  20-08-2019         2675          113430.000000   
20  Control Campaign  21-08-2019         1803           74654.000000   
21  Control Campaign  22-08-2019         2939          105705.000000   
22  Control Campaign  23-08-2019         2496          129880.000000   
23  Control Campaign  24-08-2019         1892           72515.000000   
24  Control Campaign  25-08-2019         1962          117006.000000   
25  Control Campaign  26-08-2019         2233          124897.000000   
26  Control Campaign  27-08-2019         2061          104678.000000   
27  Control Campaign  28-08-2019         2421          141654.000000   
28  Control Campaign  29-08-2019         2375           92029.000000   
29  Control Campaign  30-08-2019         2324          111306.000000   

            Reach  Number of Website Clicks  Number of Searches  \
0    56930.000000               7016.000000         2290.000000   
1   102513.000000               8110.000000         2033.000000   
2   110862.000000               6508.000000         1737.000000   
3    61235.000000               3065.000000         1042.000000   
4    88844.931034               5320.793103         2221.310345   
5    87998.000000               4028.000000         1709.000000   
6   127852.000000               2640.000000         1388.000000   
7    65217.000000               7260.000000         3047.000000   
8    94896.000000               6198.000000         2487.000000   
9    91257.000000               2277.000000         2475.000000   
10   95843.000000               8137.000000         2941.000000   
11  100189.000000               2993.000000         1397.000000   
12   68214.000000               6554.000000         2390.000000   
13  118632.000000               4521.000000         1209.000000   
14  102479.000000               4896.000000         1179.000000   
15   42859.000000               5224.000000         2427.000000   
16  106518.000000               6628.000000         1756.000000   
17   96518.000000               7253.000000         2447.000000   
18   81268.000000               3706.000000         2483.000000   
19   78625.000000               2578.000000         1001.000000   
20   59873.000000               5691.000000         2711.000000   
21   86218.000000               6843.000000         3102.000000   
22  109413.000000               4410.000000         2896.000000   
23   51987.000000               4085.000000         1274.000000   
24  100398.000000               4234.000000         2423.000000   
25   98432.000000               

In [7]:
df_Test.dtypes

Campaign Name               object
Date                        object
Spend [USD]                  int64
Number of Impressions        int64
Reach                        int64
Number of Website Clicks     int64
Number of Searches           int64
Number of View Content       int64
Number of Add to Cart        int64
Number of Purchase           int64
dtype: object

In [8]:
df_Control.dtypes

Campaign Name                object
Date                         object
Spend [USD]                   int64
Number of Impressions       float64
Reach                       float64
Number of Website Clicks    float64
Number of Searches          float64
Number of View Content      float64
Number of Add to Cart       float64
Number of Purchase          float64
dtype: object

In [9]:
# Changing data type
for column in df_Test.columns:
    if df_Test[column].dtype == 'object':
        df_Test[column] = df_Test[column].astype('string')
df_Test.dtypes

Campaign Name               string
Date                        string
Spend [USD]                  int64
Number of Impressions        int64
Reach                        int64
Number of Website Clicks     int64
Number of Searches           int64
Number of View Content       int64
Number of Add to Cart        int64
Number of Purchase           int64
dtype: object

In [10]:
# Changing data type
for column in df_Control.columns:
    if df_Control[column].dtype == 'object':
        df_Control[column] = df_Control[column].astype('string')
df_Control.dtypes

Campaign Name                string
Date                         string
Spend [USD]                   int64
Number of Impressions       float64
Reach                       float64
Number of Website Clicks    float64
Number of Searches          float64
Number of View Content      float64
Number of Add to Cart       float64
Number of Purchase          float64
dtype: object

In [11]:
# Changing data type
for column in df_Control.columns:
    if df_Control[column].dtype == 'float64':
        df_Control[column] = df_Control[column].astype('int64')
df_Control.dtypes

Campaign Name               string
Date                        string
Spend [USD]                  int64
Number of Impressions        int64
Reach                        int64
Number of Website Clicks     int64
Number of Searches           int64
Number of View Content       int64
Number of Add to Cart        int64
Number of Purchase           int64
dtype: object

In [12]:
df_Control

Campaign Name        Date  Spend [USD]  Number of Impressions   Reach  \
0   Control Campaign  01-08-2019         2280                  82702   56930   
1   Control Campaign  02-08-2019         1757                 121040  102513   
2   Control Campaign  03-08-2019         2343                 131711  110862   
3   Control Campaign  04-08-2019         1940                  72878   61235   
4   Control Campaign  05-08-2019         1835                 109559   88844   
5   Control Campaign  06-08-2019         3083                 109076   87998   
6   Control Campaign  07-08-2019         2544                 142123  127852   
7   Control Campaign  08-08-2019         1900                  90939   65217   
8   Control Campaign  09-08-2019         2813                 121332   94896   
9   Control Campaign  10-08-2019         2149                 117624   91257   
10  Control Campaign  11-08-2019         2490                 115247   95843   
11  Control Campaign  12-08-2019         2319                 116639  100189   
12  Control Campaign  13-08-2019         2697                  82847   68214   
13  Control Campaign  14-08-2019         1875                 145248  118632   
14  Control Campaign  15-08-2019         2774                 132845  102479   
15  Control Campaign  16-08-2019         2024                  71274   42859   
16  Control Campaign  17-08-2019         2177                 119612  106518   
17  Control Campaign  18-08-2019         1876                 108452   96518   
18  Control Campaign  19-08-2019         2596                 107890   81268   
19  Control Campaign  20-08-2019         2675                 113430   78625   
20  Control Campaign  21-08-2019         1803                  74654   59873   
21  Control Campaign  22-08-2019         2939                 105705   86218   
22  Control Campaign  23-08-2019         2496                 129880  109413   
23  Control Campaign  24-08-2019         1892                  72515   51987   
24  Control Campaign  25-08-2019         1962                 117006  100398   
25  Control Campaign  26-08-2019         2233                 124897   98432   
26  Control Campaign  27-08-2019         2061                 104678   91579   
27  Control Campaign  28-08-2019         2421                 141654  125874   
28  Control Campaign  29-08-2019         2375                  92029   74192   
29  Control Campaign  30-08-2019         2324                 111306   88632   

    Number of Website Clicks  Number of Searches  Number of View Content  \
0                       7016                2290                    2159   
1                       8110                2033                    1841   
2                       6508                1737                    1549   
3                       3065                1042                     982   
4                       5320                2221                    1943   
5                       4028                1709                    1249   
6                       2640                1388                    1106   
7                       7260                3047                    2746   
8                       6198                2487                    2179   
9                       2277                2475                    1984   
10                      8137                2941                    2486   
11                      2993                1397                    1147   
12                      6554                2390                    1975   
13                      4521                1209                    1149   
14                      4896                1179                    1005   
15                      5224                2427                    2158   
16                      6628                1756                    1642   
17                      7253                2447                    2115   
18                      3706                2483                    2098   
19    

In [35]:
# Concatinating both the data frames
df = pd.concat([df_Control,df_Test],axis = 0,ignore_index=True)
df.dtypes

Campaign Name               string
Date                        string
Spend [USD]                  int64
Number of Impressions        int64
Reach                        int64
Number of Website Clicks     int64
Number of Searches           int64
Number of View Content       int64
Number of Add to Cart        int64
Number of Purchase           int64
dtype: object

In [14]:
print(df["Campaign Name"].value_counts())

Control Campaign    30
Test Campaign       30
Name: Campaign Name, dtype: Int64


In [15]:
# The data set that I downloaded has some wrong data ,that is some rows have Number of add to cart values greater than 
# Number of View Content ,which is shown below
df[df["Number of View Content"] < df ["Number of Add to Cart"]]

Campaign Name        Date  Spend [USD]  Number of Impressions   Reach  \
3   Control Campaign  04-08-2019         1940                  72878   61235   
6   Control Campaign  07-08-2019         2544                 142123  127852   
11  Control Campaign  12-08-2019         2319                 116639  100189   
13  Control Campaign  14-08-2019         1875                 145248  118632   
14  Control Campaign  15-08-2019         2774                 132845  102479   
19  Control Campaign  20-08-2019         2675                 113430   78625   
27  Control Campaign  28-08-2019         2421                 141654  125874   
34     Test Campaign  05-08-2019         2297                 114295   95138   

    Number of Website Clicks  Number of Searches  Number of View Content  \
3                       3065                1042                     982   
6                       2640                1388                    1106   
11                      2993                1397                    1147   
13                      4521                1209                    1149   
14                      4896                1179                    1005   
19                      2578                1001                     848   
27                      6287                1672                    1589   
34                      5863                2106                     858   

    Number of Add to Cart  Number of Purchase  
3                    1183                 340  
6                    1166                 499  
11                   1439                 794  
13                   1339                 788  
14                   1641                 366  
19                   1709                 299  
27                   1711                 643  
34                    956                 768

In [16]:
# So just for this project lets swap the values df["Number of View Content"] and df ["Number of Add to Cart"] wherever this 
# condition satisfies - "df["Number of View Content"] < df ["Number of Add to Cart"]"
import pandas as pd
pd.options.mode.chained_assignment = None

for i in range(df.shape[0]):
    if(df["Number of View Content"][i] < df["Number of Add to Cart"][i]):
        temp = df["Number of Add to Cart"][i]
        df["Number of Add to Cart"][i] = df["Number of View Content"][i]
        df["Number of View Content"][i] = temp
df[df["Number of View Content"] < df ["Number of Add to Cart"]]

Empty DataFrame
Columns: [Campaign Name, Date, Spend [USD], Number of Impressions, Reach, Number of Website Clicks, Number of Searches, Number of View Content, Number of Add to Cart, Number of Purchase]
Index: []

## Control campaign is pertforming better than Test campaign in getting more impressoins in less cost

In [78]:
import plotly.express as px
figure = px.scatter(data_frame = df,
                    x = "Spend [USD]",
                    y = "Number of Impressions",
                    size = "Spend [USD]",
                    color = "Campaign Name",
                    trendline="ols")
figure.show()

## Test campaign has more number of searches than Control campaign

In [18]:
import plotly.express as px
# fig = px.pie(df, values="Number of Searches", names='Campaign Name', color_discrete_sequence=px.colors.sequential.RdBu)
fig = px.pie(df, values="Number of Searches", names='Campaign Name')
fig.show()

## Test campaign has more number of Website Clicks than Control campaign

In [19]:
# fig = px.pie(df, values="Number of Searches", names='Campaign Name', color_discrete_sequence=px.colors.sequential.RdBu)
fig = px.pie(df, values="Number of Website Clicks", names='Campaign Name')
fig.show()

## Control campaign has more number of Content views than Test campaign

In [20]:
fig = px.pie(df, values="Number of View Content", names='Campaign Name')
fig.show()

## Both Test campaign and  Control campaign almost has the same Number of Purchases

In [21]:
fig = px.pie(df, values="Number of Purchase", names='Campaign Name')
fig.show()

## Add to cart by content viewed plot for both the campaigns are almost similar, but at the right end of the plot, lightblue dots of Control campaign shows that conversion rate(Add to cart/Content viewed) of Control campaign is little higher than Test campaign.

In [22]:
figure = px.scatter(data_frame = df, 
                    x="Number of Add to Cart",
                    y="Number of View Content", 
                    size="Number of Add to Cart", 
                    color= "Campaign Name", 
                    trendline="ols")
figure.show()

## Number of Purchases by Add to cart scatter plot shows that shows that conversion rate(Purchases/Add to cart) of Test campaign is higher than Control campaign.

In [23]:
figure = px.scatter(data_frame = df, 
                    x="Number of Purchase",
                    y="Number of Add to Cart", 
                    size="Number of Purchase", 
                    color= "Campaign Name", 
                    trendline="ols")# Plotly Express allows you to add Ordinary Least Squares regression trendline to scatterplots with the trendline argument.
figure.show()

In [45]:
# Trendline charts for Test Campaign
for column in df.columns:
    if column not in ["Campaign Name","Date"]:
        fig = px.line(df_Test, x="Date", y=column, title=  column + " by Date")  
        fig.show()

In [44]:
# Trendline charts for Control Campaign
for column in df.columns:
    if column not in ["Campaign Name","Date"]:
        fig = px.line(df_Control, x="Date", y=column, title=  column + " by Date")  
        fig.show()

## Add Spend per Purchase

In [24]:
Control_AddSpendperPurchase = sum(df[df["Campaign Name"] == "Control Campaign"]["Spend [USD]"])/sum(df[df["Campaign Name"] == "Control Campaign"]["Number of Purchase"]) 
print("ControlCampaign_AddSpendperPurchase "+ str(round(Control_AddSpendperPurchase,2)) + " USD")

Test_AddSpendperPurchase = sum(df[df["Campaign Name"] == "Test Campaign"]["Spend [USD]"])/sum(df[df["Campaign Name"] == "Test Campaign"]["Number of Purchase"]) 
print("TestCampaign_AddSpendperPurchase    " + str(round(Test_AddSpendperPurchase,2)) + " USD")

ControlCampaign_AddSpendperPurchase 4.38 USD
TestCampaign_AddSpendperPurchase    4.92 USD


## Add Spend per Click

In [76]:
Control_AddSpendperClicks = sum(df[df["Campaign Name"] == "Control Campaign"]["Spend [USD]"])/sum(df[df["Campaign Name"] == "Control Campaign"]["Number of Website Clicks"]) 
print("ControlCampaign_AddSpendperClicks "+ str(round(Control_AddSpendperClicks,2)) + " USD")

Test_AddSpendperClicks = sum(df[df["Campaign Name"] == "Test Campaign"]["Spend [USD]"])/sum(df[df["Campaign Name"] == "Test Campaign"]["Number of Website Clicks"]) 
print("TestCampaign_AddSpendperClicks    " + str(round(Test_AddSpendperClicks,2)) + " USD")

ControlCampaign_AddSpendperClicks 0.43 USD
TestCampaign_AddSpendperClicks    0.42 USD


## Conversion Rate (Purchase per Click)

In [26]:
Control_PurchaseperClicks = sum(df[df["Campaign Name"] == "Control Campaign"]["Number of Purchase"])/sum(df[df["Campaign Name"] == "Control Campaign"]["Number of Website Clicks"]) 
print("ControlCampaign_PurchaseperClicks "+ str(round(Control_PurchaseperClicks,2)))

Test_PurchaseperClicks = sum(df[df["Campaign Name"] == "Test Campaign"]["Number of Purchase"])/sum(df[df["Campaign Name"] == "Test Campaign"]["Number of Website Clicks"]) 
print("TestCampaign_PurchaseperClicks    " + str(round(Test_PurchaseperClicks,2)))

ControlCampaign_PurchaseperClicks 0.1
TestCampaign_PurchaseperClicks    0.09


### Conversion Rate for Control campaign is higher than Test campaign 

## Click Through Rate(Click per Impressions)

In [28]:
Control_CTR = sum(df[df["Campaign Name"] == "Control Campaign"]["Number of Website Clicks"])/sum(df[df["Campaign Name"] == "Control Campaign"]["Number of Impressions"]) 
print("ControlCampaign_CTR "+ str(round(Control_CTR,2)))

Test_CTR = sum(df[df["Campaign Name"] == "Test Campaign"]["Number of Website Clicks"])/sum(df[df["Campaign Name"] == "Test Campaign"]["Number of Impressions"]) 
print("TestCampaign_CTR    " + str(round(Test_CTR,2)))

ControlCampaign_CTR 0.05
TestCampaign_CTR    0.08


In [74]:
import warnings
warnings.simplefilter(action='ignore', category=FutureWarning) #To supress future warning

Control_sum = df_Control.sum()
Test_sum = df_Test.sum()
print("Control Campaign\n")
print(Control_sum)

print("\nTest Campaign\n")
print(Test_sum)

Control Campaign

Spend [USD]                   68653
Number of Impressions       3286792
Reach                       2665347
Number of Website Clicks     159623
Number of Searches            66639
Number of View Content        58313
Number of Add to Cart         39000
Number of Purchase            15683
dtype: int64

Test Campaign

Spend [USD]                   76892
Number of Impressions       2237544
Reach                       1604747
Number of Website Clicks     180970
Number of Searches            72569
Number of View Content        55740
Number of Add to Cart         26446
Number of Purchase            15637
dtype: int64


# Important Points

## 1. Control campaign is better in reaching more users and providing more engagement in less cost than Test Campaign.

## 2. Click Through Rate for test campaign (8%) is higher than Control Campaign (5%)

## 3. Ad Spend per Purchase is less for Control Campaign (4.38 USD) than Test Campaign                                                       (4.92 USD)

# Conclusion

## There are more number of purchases and reach for control campaign which has created more sales than test campaign but even though engagement and reach is less for test campaign it has good conversion rate it means test campaign add is reaching the relevant audience. 

## Control Campaign : Where more reach and engagenment is required.
## Test Campaign       : Where specific audience targetted